ticketutil/servicenow.py

```
@@ -0,0 +1,321 @@
+        self.sys_id = result['sys_id']
+        self.ticket_url = self._generate_ticket_url()
+        logging.info('Created issue {0} - {1}'.format(self.ticket_id,
+                                                      self.ticket_url))
+
+    def change_status(self, status):
+        """
+        Change ServiceNow ticket status
+
+        :param status: State to change to
+        """
+        if not self.sys_id:
+            logging.error('No ticket ID associated with ticket object. Set '
+                          'ticket ID with set_ticket_id(ticket_id)')
+            return
+
+        try:
+            logging.info('Changing ticket status')
+            fields = {'state': status}
+            self.edit(**fields)
+        except:
+            logging.error('Failed to change ticket status')
+
+    def edit(self, **kwargs):
+        """
+        Edit ticket
+
+        Edits a ServiceNow ticket, ticked_id (sys_id) must be set beforehand.
+        You can set ticket_id by calling set_ticket_id(ticket_id) method.
+
+        :param kwargs: optional fields
+
+        Fields example:
+        'contact_type' : 'Email',
+        'u_opened_for' : 'PNT',
+        'assigned_to' : 'pzubaty',
+        'impact' : '2',
+        'urgency' : '2',
+        'priority' : '2'
+        """
+        if not self.sys_id:
+            logging.error('No ticket ID associated with ticket object. Set '
+                          'ticket ID with set_ticket_id(ticket_id)')
+            return
+        params = ''
+        for key, value in kwargs.items():
+            params += ', "{}" : "{}"'.format(key, value)
+        params = '{' + params[1:] + '}'
+
+        url = self.rest_url + '/' + self.sys_id
+        result = self.api_put(url, params)
+        if result['number']:
+            logging.info('Issue edited {0} - {1}'
+                         .format(self.ticket_id, self.ticket_url))
+        else:
+            logging.error('Error while editing issue')
```

More silenced errors. If you fix the `api_*` methods to raise errors properly then this `if/else` block will not be necessary.